bst.py

```diff
@@ -0,0 +1,129 @@
+# -*- coding: utf-8 -*-
+from __future__ import unicode_literals
+import random
+import timeit
+# source of info http://interactivepython.org/runestone/static/pythonds/Trees/bst.html
+
+
+class Node(object):
+    def __init__(self, val, left=None, right=None, parent=None):
+        """Instantiates a node"""
+        self.val = val
+        self.left = left
+        self.right = right
+        self.parent = parent
+
+    def balance(self):
+        if self.right.depth() > self.left.depth():
+            return 1
+        elif self.left.depth() > self.right.depth():
+            return -1
+        else:
+            return 0
```

Comment:
In our case, we actually want to get the difference between the left and right depths, not a 1, -1, 0 value.